services/f-droid/f-droid.service.js

```diff
@@ -42,9 +54,30 @@ module.exports = class FDroid extends BaseService {
     }
   }
 
-  async handle({ appId }) {
-    const result = await this.fetch({ appId })
-    return this.constructor.render(result)
+  async handle({ appId }, queryParams) {
+    const constructor = this.constructor
+    const { metadata_format: format } = constructor.validateParams(queryParams)
+    const result = await this.fetch({ appId, format })
+
+    return constructor.render(result)
+  }
+
+  static validateParams(queryParams) {
+    const queryParamsSchema = Joi.object({
+      metadata_format: Joi.string().valid(['yml', 'txt']),
+    }).required()
+
+    return validate(
```

Comment:
Just a heads up, particularly to the other maintainers, that I've moved this function into `BaseService` in #2399. Presumably this PR will land first and the duplication can be removed in #2399, or later. The changes don't conflict.

Reply:
done https://github.com/badges/shields/pull/2377/commits/5156c821e0669b8c1918a5293069349734eafb8d